src/main/scala/tilelink/Parameters.scala

```diff
@@ -876,62 +912,92 @@ class TLMasterPortParameters private(
-  val anySupportLogical    = clients.map(!_.supportsLogical.none)   .reduce(_ || _)
-  val anySupportGet        = clients.map(!_.supportsGet.none)       .reduce(_ || _)
-  val anySupportPutFull    = clients.map(!_.supportsPutFull.none)   .reduce(_ || _)
-  val anySupportPutPartial = clients.map(!_.supportsPutPartial.none).reduce(_ || _)
-  val anySupportHint       = clients.map(!_.supportsHint.none)      .reduce(_ || _)
-
-  // These return Option[TLClientParameters] for your convenience
-  def find(id: Int) = clients.find(_.sourceId.contains(id))
+  // Operation sizes supported by all inward Masters
+  val allSupportProbe      = masters.map(_.supportsProbe)     .reduce(_ intersect _)
+  val allSupportArithmetic = masters.map(_.supportsArithmetic).reduce(_ intersect _)
+  val allSupportLogical    = masters.map(_.supportsLogical)   .reduce(_ intersect _)
+  val allSupportGet        = masters.map(_.supportsGet)       .reduce(_ intersect _)
+  val allSupportPutFull    = masters.map(_.supportsPutFull)   .reduce(_ intersect _)
+  val allSupportPutPartial = masters.map(_.supportsPutPartial).reduce(_ intersect _)
+  val allSupportHint       = masters.map(_.supportsHint)      .reduce(_ intersect _)
+
+  // Operation is supported by at least one master
+  val anySupportProbe      = masters.map(!_.supportsProbe.none)     .reduce(_ || _)
+  val anySupportArithmetic = masters.map(!_.supportsArithmetic.none).reduce(_ || _)
+  val anySupportLogical    = masters.map(!_.supportsLogical.none)   .reduce(_ || _)
+  val anySupportGet        = masters.map(!_.supportsGet.none)       .reduce(_ || _)
+  val anySupportPutFull    = masters.map(!_.supportsPutFull.none)   .reduce(_ || _)
+  val anySupportPutPartial = masters.map(!_.supportsPutPartial.none).reduce(_ || _)
+  val anySupportHint       = masters.map(!_.supportsHint.none)      .reduce(_ || _)
+
+  // These return Option[TLMasterParameters] for your convenience
+  def find(id: Int) = masters.find(_.sourceId.contains(id))
 
   // Synthesizable lookup methods
-  def find(id: UInt) = Vec(clients.map(_.sourceId.contains(id)))
+  def find(id: UInt) = Vec(masters.map(_.sourceId.contains(id)))
   def contains(id: UInt) = find(id).reduce(_ || _)
 
-  def requestFifo(id: UInt) = Mux1H(find(id), clients.map(c => Bool(c.requestFifo)))
+  def requestFifo(id: UInt) = Mux1H(find(id), masters.map(c => Bool(c.requestFifo)))
 
-  private def safety_helper(member: TLClientParameters => TransferSizes)(id: UInt, lgSize: UInt) = {
-    val allSame = clients.map(member(_) == member(clients(0))).reduce(_ && _)
-    if (allSame) member(clients(0)).containsLg(lgSize) else {
-      Mux1H(find(id), clients.map(member(_).containsLg(lgSize)))
+  // Available during RTL runtime, checks to see if (id, size) is supported by the master's (master's)  diplomatic parameters
+  private def safety_helper(member: TLMasterParameters => TransferSizes)(id: UInt, lgSize: UInt) = {
```

Comment:
We need to reason about which features of the proposed`emitHelper` need to be merged with this function and whether they actually produce different logic in valuable ways.

I recommend changing the names to be `addressHelper` and `sourceIdHelper`, since I believe the supports/emits aspect is orthogonal (i.e. it is whichever `TLMasterParameters => TransferSizes` you happen to pass in, which could be either about supporting or about emitting).